spyder/utils/sourcecode.py

```diff
@@ -120,12 +125,99 @@ def get_identifiers(source_code):
+    """
+    components = []
+    # The loop guarantees that the returned components can be
+    # os.path.joined with the path separator and point to the same
+    # location:    
+    while True:
+        (new_path, tail) = os.path.split(path)  # Works on any platform
+        components.append(tail)        
+        if new_path == path:  # Root (including drive, on Windows) reached
+            break
+        path = new_path
+    components.append(new_path)    
+    components.reverse()  # First component first 
+    return components
+
+def differentiate_prefix(path_components0, path_components1):
+    """
+    Return the differentiated prefix of the given two iterables. 
+     
+    Taken from http://stackoverflow.com/q/21498939/438386
+    """
+    longest_prefix = []
+    root_comparison = False
+    common_elmt = None
+    for index, (elmt0, elmt1) in enumerate(zip(path_components0, path_components1)):
+        if elmt0 != elmt1:
+            if index == 2:
+                root_comparison = True
+            break
+        else:
+            common_elmt = elmt0
+        longest_prefix.append(elmt0)
+    file_name_length = len(path_components0[len(path_components0) - 1])
+    path_0 = os.path.join(*path_components0)[:-file_name_length - 1]
+    if len(longest_prefix) > 0:
+        longest_path_prefix = os.path.join(*longest_prefix)
+        longest_prefix_length = len(longest_path_prefix) + 1
+        if path_0[longest_prefix_length:] != '' and not root_comparison:
+            path_0 = path_0[longest_prefix_length:]
+        elif not root_comparison:
+            path_0 = common_elmt
+        elif sys.platform.startswith('linux') and path_0 == '':
+            path_0 = '/'
+    return path_0
+
+def get_file_title(data, index):
+    """Get tab title without ambiguation."""
+    finfo = data[index]
+    fname = os.path.basename(finfo.filename)
+    same_name_files = get_same_name_files(data, fname)
+    if len(same_name_files) > 1:
+        compare_path = shortest_path(same_name_files)
+        if compare_path == finfo.filename:
+            same_name_files.pop(index)
+            compare_path = shortest_path(same_name_files)    
+        diff_path = differentiate_prefix(path_components(finfo.filename),
+                                             path_components(compare_path))
+        diff_path_length = len(diff_path)
+        path_component = path_components(diff_path)
+        if (diff_path_length > 20 and len(path_component) > 2):
+            if path_component[0] != '/':
+                path_component = [path_component[0], '...', 
+                                          path_component[-1]]
+            else:
+                path_component = [path_component[2], '...', 
+                                          path_component[-1]]
+            diff_path = os.path.join(*path_component)
+        fname = fname + " - " + diff_path    
+    return fname
+
+def get_same_name_files(data, fname):

```

Comment:
Please also make this function independent of data (or move it to the Editor :-)